File: 13-clients/sdk/swift-etrid-sdk/Sources/EtridSDK/Account.swift
/// Account management and cryptographic operations

import Foundation
import Crypto

/// Ëtrid account
public struct Account {
    private let privateKey: Curve25519.Signing.PrivateKey

    private init(privateKey: Curve25519.Signing.PrivateKey) {
        self.privateKey = privateKey
    }

    /// Create account from mnemonic phrase
    /// - Parameter mnemonic: 12 or 24 word mnemonic phrase
    /// - Returns: New Account instance
    public static func fromMnemonic(_ mnemonic: String) throws -> Account {
        // Simplified implementation - real version would use BIP39
        let privateKey = Curve25519.Signing.PrivateKey()
        return Account(privateKey: privateKey)
    }

    /// Generate a new random account
    /// - Returns: New Account instance
    public static func generate() -> Account {
        let privateKey = Curve25519.Signing.PrivateKey()
        return Account(privateKey: privateKey)
    }

    /// Get the account address (SS58 format)
    public var address: String {
        // Placeholder - real implementation would encode properly
        return publicKey
    }

    /// Get the public key (hex format)
    public var publicKey: String {
        privateKey.publicKey.rawRepresentation.map { String(format: "%02x", $0) }.joined()
    }

    /// Sign a message
    /// - Parameter message: Message to sign
    /// - Returns: Signature data
    public func sign(_ message: Data) throws -> Data {
        try privateKey.signature(for: message)
    }
}

File: 13-developer-tools/sdk/python-etrid-sdk/etrid_sdk/account.py
"""Account management and cryptographic operations"""

from substrateinterface import Keypair

class Account:
    """Ëtrid account"""

    def __init__(self, keypair: Keypair):
        self._keypair = keypair

    @classmethod
    def from_mnemonic(cls, mnemonic: str) -> "Account":
        """
        Create account from mnemonic phrase

        Args:
            mnemonic: 12 or 24 word mnemonic phrase

        Returns:
            New Account instance
        """
        keypair = Keypair.create_from_mnemonic(mnemonic)
        return cls(keypair)

    @classmethod
    def generate(cls) -> "Account":
        """
        Generate a new random account

        Returns:
            New Account instance
        """
        keypair = Keypair.create_from_mnemonic(Keypair.generate_mnemonic())
        return cls(keypair)

    @property
    def address(self) -> str:
        """Get the account address (SS58 format)"""
        return self._keypair.ss58_address

    @property
    def public_key(self) -> str:
        """Get the public key (hex format)"""
        return self._keypair.public_key.hex()

    def sign(self, message: bytes) -> bytes:
        """
        Sign a message

        Args:
            message: Message to sign

        Returns:
            Signature bytes
        """
        return self._keypair.sign(message)
